controllers/provisioners/eks/cloud.go

```diff
@@ -238,7 +238,7 @@ func (ctx *EksInstanceGroupContext) CloudDiscovery() error {
 
 	if spec.IsLaunchTemplate() {
 		state.ScalingConfiguration, err = scaling.NewLaunchTemplate(instanceGroup.NamespacedName(), ctx.AwsWorker, &scaling.DiscoverConfigurationInput{
-			ScalingGroup: targetScalingGroup,
```

Comment:
I think the current implementation discovers the active launch template by the scaling group name, can you look inside NewLaunchTemplate and see if there is a reason this is not happening?

Reply:
Looks like we are not setting TargetConfigName inside Discover:
https://github.com/sahilbadla/instance-manager/blob/58ad1a253a80900c2a1047c5809ca1d94e300d00/controllers/provisioners/eks/scaling/launchtemplate.go#L58

Can you try that instead? 
That way we don't rely on status field, but rather discover the active Launch Template name each time from the scaling group name.